### PR TITLE
KalturaLogger logs: settings > custom data module

### PR DESCRIPTION
**Pull request by diamond-darrell**
### PR information

**What is the current behavior?**

**What is the new behavior?**
Utilize KalturaLogger to log settings > custom data module

**Does this PR introduce a breaking change?**

---
This change is [Reviewable](https://reviewable.io/reviews/kaltura/kmc-ng/553)

### CODE REVIEW

**Comment by esakal** [comment]
Review status: 0 of 4 files reviewed at latest revision, all discussions resolved.

---

*[src/applications/settings-custom-data-app/schemas/custom-schema/custom-schema.component.ts, line 54 at r1](https://beta.reviewable.io/reviews/kaltura/kmc-ng/553#-L9CFZO7EQhOMKxVeG1o:-L9CFZO7EQhOMKxVeG1p:bdz00tc) ([raw file](https://github.com/kaltura/kmc-ng/blob/f7d89b8bdfedc3dbea383fbc3aa1000a7296507d/src/applications/settings-custom-data-app/schemas/custom-schema/custom-schema.component.ts#L54)):*
> ```XML
>       this._title = this._appLocalization.get('applications.settings.metadata.editCustomSchema');
>     } else {
>       this._logger.info(`enter edit schema mode`);
> ```

shouldn't it be `enter add new schema mode`

---

*[src/applications/settings-custom-data-app/schemas/custom-schema/custom-schema.component.ts, line 200 at r1](https://beta.reviewable.io/reviews/kaltura/kmc-ng/553#-L9CFztr6HcO772sa_dC:-L9CFztr6HcO772sa_dD:brzii20) ([raw file](https://github.com/kaltura/kmc-ng/blob/f7d89b8bdfedc3dbea383fbc3aa1000a7296507d/src/applications/settings-custom-data-app/schemas/custom-schema/custom-schema.component.ts#L200)):*
> ```XML
> 
>   public _bulkRemove(): void {
>     this._logger.info(`handle 'bulk remove fields' action by the user`);
> ```

I'm thinking out load, the logs should tell a story about what happened. Since we add it in retrospect we need to be careful not relying on the code to provide the context since the logs will be read separately. 

This is what the PR currently write to the log:
- handle 'bulk remove fields' action by the user
- handle accept 'bulk remove fields' action by the user
**-or-**
- handle 'bulk remove fields' action by the user
- handle reject 'bulk remove fields' action by the user

see the following suggestion:
- handle action 'bulk remove fields' requested by the user. show confirmation dialog
- user confirmed, continue action
**-or-**
- handle action 'bulk remove fields' requested by the user. show confirmation dialog
- user didn't confirm, abort action

you might think it is minor but I really think the latter better describe the behavior of the application. To workaround the code trap, I suggest you to run the scenario and read the logs in the browser console and compare them to the application behavior and see if things are missing or need to be more concrete. 

What do you think?

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/kaltura/kmc-ng/553)*

**Comment by esakal** [comment]
Reviewed 4 of 4 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/kaltura/kmc-ng/553)*

**Comment by diamond-darrell** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[src/applications/settings-custom-data-app/schemas/custom-schema/custom-schema.component.ts, line 54 at r1](https://beta.reviewable.io/reviews/kaltura/kmc-ng/553#-L9CFZO7EQhOMKxVeG1o:-L9ETeBq6tnApwqdiNvd:bljxyd3) ([raw file](https://github.com/kaltura/kmc-ng/blob/f7d89b8bdfedc3dbea383fbc3aa1000a7296507d/src/applications/settings-custom-data-app/schemas/custom-schema/custom-schema.component.ts#L54)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

shouldn't it be `enter add new schema mode`
</blockquote></details>

yep, typo

---

*[src/applications/settings-custom-data-app/schemas/custom-schema/custom-schema.component.ts, line 200 at r1](https://beta.reviewable.io/reviews/kaltura/kmc-ng/553#-L9CFztr6HcO772sa_dC:-L9ETlHc9mflMQwta_F7:b-9ni7z1) ([raw file](https://github.com/kaltura/kmc-ng/blob/f7d89b8bdfedc3dbea383fbc3aa1000a7296507d/src/applications/settings-custom-data-app/schemas/custom-schema/custom-schema.component.ts#L200)):*
<details><summary><i>Previously, esakal (Eran Sakal) wrote…</i></summary><blockquote>

I'm thinking out load, the logs should tell a story about what happened. Since we add it in retrospect we need to be careful not relying on the code to provide the context since the logs will be read separately. 

This is what the PR currently write to the log:
- handle 'bulk remove fields' action by the user
- handle accept 'bulk remove fields' action by the user
**-or-**
- handle 'bulk remove fields' action by the user
- handle reject 'bulk remove fields' action by the user

see the following suggestion:
- handle action 'bulk remove fields' requested by the user. show confirmation dialog
- user confirmed, continue action
**-or-**
- handle action 'bulk remove fields' requested by the user. show confirmation dialog
- user didn't confirm, abort action

you might think it is minor but I really think the latter better describe the behavior of the application. To workaround the code trap, I suggest you to run the scenario and read the logs in the browser console and compare them to the application behavior and see if things are missing or need to be more concrete. 

What do you think?

</blockquote></details>

Basically, that's what I was doing: running the app, going through the flow and add missing logs. I just thought you want to use same statements for logging so they all will look alike, and I took `edit-role` logs as an example, but I hope I've got your point 🙂

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/kaltura/kmc-ng/553)*
